### DISCUSSION TITLE
Using Intents for Non-Visual Actions

### DISCUSSION THREAD

**Discussion by hampshan** [finos/FDC3#676] [Q&A] [answered]
I am thinking about using an Intent to perform an action that does not require the launch of a new panel, nor does it need to send information to a different panel. The scenario is similar to calling a module to perform a calculation. 

Ideally, I'd like to register the module as an app so that the action is auto-discovered and/or auto loaded by other apps using findIntent. 

Is there a recommended way to do this? I think the current App Directory app definition is really targeted at visual components, or at least actions that should shows up in an App Store.

**Comment by kriswest** [comment, marked as answer]
We regularly work with service applications - one of the reasons we're particularly keen on the 2.0 API changes to allow the return of results (so they can essentially provide something more like an API than a UI).

Although the AppD does 'feel' like its intended for apps with a UI/show a window, I don't think anything actually requires it and it should still work for your non-visual use-cases. I also think there is a use-case for non-visual apps to show up in an intent resolver when they need to (e.g. what service to get a price, or some other datum, from) - also worth remembering that the resolver shouldn't show up and should auto-resolve if only one option for the intent/context pairing is available. As for showing up in an app-store... you might still want it to I guess - although some way to indicate that its non-visual might be useful (perhaps also useful in the intent resolver?). 

If you really don't want it to show up, we'd need a new field on the record to indicate that. I think that related to this issue I raised last week that we've been thinking about for a while:
- https://github.com/finos/FDC3/issues/674

It wouldn't actually prevent it showing up / indicate that it shouldn't show up, but might provide a mechanism to pull it in if we added something to stop it showing up.

@hampshan hopefully that gives you something to think about. We can get this onto the next AppD meeting agenda if you want to discuss further (it'd be great if you could raise an issue with a what you'd like to focus the discussion on if so).

**Reply by hampshan** [reply]
@kriswest Thanks. This makes sense, I do think that a new field in the App Directory would help for this type of use-case. I'll raise an issue.
